lib/src/main/java/graphql/nadel/engine/NadelIncrementalResultSupport.kt

```diff
@@ -0,0 +1,181 @@
+package graphql.nadel.engine
+
+import graphql.incremental.DelayedIncrementalPartialResult
+import graphql.nadel.engine.NadelIncrementalResultSupport.OutstandingJobCounter.OutstandingJobHandle
+import graphql.nadel.engine.util.copy
+import graphql.nadel.util.getLogger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.launch
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+ * todo: we do not handle the case where defer jobs finish before [onInitialResultComplete]
+ */
+class NadelIncrementalResultSupport internal constructor(
+    private val delayedResultsChannel: Channel<DelayedIncrementalPartialResult> = Channel(
+        capacity = 100,
+        onBufferOverflow = BufferOverflow.DROP_LATEST,
+        onUndeliveredElement = {
+            log.error("Dropping incremental result because of buffer overflow")
+        },
+    ),
```

Comment:
Default channel buffers 100 elements if nobody is consuming them.

Should be high enough.